src/main/java/io/stargate/sgv2/jsonapi/service/shredding/collections/DocumentShredderListener.java

```diff
@@ -27,5 +27,7 @@ public interface DocumentShredderListener {
 
   void shredVector(JsonPath path, ArrayNode vector);
 
+  void shredVector(JsonPath path, ObjectNode vector);
```

Comment:
Instead of passing the whole `ObjectNode`, should only pass `TextNode` value, or maybe just `String` contained. Caller can and should check it is textual (if (node.isTextual()). That way called method need not try to go and find it again.